### PR TITLE
Start migration to Readium 2 alpha and prepare DRM support

### PR DESCRIPTION
**Pull request by qnga**

### CODE REVIEW

**Review by io7m — COMMENTED** [review]
This all looks fine to me. One question: Do we know which Readium 2 version is the one we'll be targeting? Right now `nypl-SNAPSHOT` could mean anything. :wink:

**Comment by qnga** [comment]
Sure, my `nypl` branch follows Readium `alpha` branch. Sometimes it's slightly ahead though. Besides, a few months ago there were jitpack build issues when using directly `alpha`, so I had to deactivate some checks in `build.gradle`.
A tagged version should be released in the next few months, you might stick to it for a while then.

File: org.librarysimplified.r2.vanilla/src/main/java/org/librarysimplified/r2/vanilla/internal/SR2Controller.kt [hunk 1]
```diff
@@ -537,7 +531,7 @@ internal class SR2Controller private constructor(
 
   override val bookMetadata: SR2BookMetadata =
     SR2BookMetadata(
-      id = this.publication.metadata.identifier,
+      id = this.publication.metadata.identifier!!, //FIXME : identifier is not mandatory in RWPM.
```
Comment (twaddington) [inline org.librarysimplified.r2.vanilla/src/main/java/org/librarysimplified/r2/vanilla/internal/SR2Controller.kt:534]: What is RWPM?

**Review by twaddington — APPROVED** [review]
Looks good.

File: org.librarysimplified.r2.vanilla/src/main/java/org/librarysimplified/r2/vanilla/internal/SR2Controller.kt (hunk 1 above)
Comment (qnga) [reply org.librarysimplified.r2.vanilla/src/main/java/org/librarysimplified/r2/vanilla/internal/SR2Controller.kt:534]: Readium WebPub Manifest

**Pull request merged**